src/Tokenizer.php

```diff
@@ -22,62 +23,57 @@
 final class Tokenizer
 {
     /**
-     * Reserved words (for syntax highlighting)
+     * Reserved MySQL and MariaDB keywords
```

Comment:
When reading that, one might think they shouldn't add keywords for any other DB… is that the case?

Reply:
Yes - and this is also strictly tested. The reason is we must be sure the "reserved" keywords are really "reserved", as only these keywords can be uppercased safely - for #32. All other keywords must be put into "non-reserved" list, even if they are "reserved" in other than MySQL/SQLite DB vendor.

In practise, this is very very clever, as generic SQL keywords (like `WHERE`, `UPDATE`, ...) are reserved everywhere and exotic keywords can be still highlighted using the "non-reserved" list but case is kept to prevent #95.

Reply:
Are there keywords that could be reserved according to MariaDB, but not according to other vendors? I don't get why MySQL/MariaDB is privileged.

Reply:
The reasons are historical, jdorn library was based on MySQL keywords solely.

I already added full SQLite reserved keywords list to the tests so all of them are newly contained in the "non-reserved" list.

This PR aims to add MySQL/MariaDB/SQLite keywords. Other vendors can be added later.

For highlighting, "reserved"/"non-reserved" lists are used the same way. For #32, we need "reserved" list to be minimal subset (intersect) of all vendors.

Reply:
> Are there keywords that could be reserved according to MySQL/MariaDB, but not according to other vendors? If yes, what did you do with these keywords?

Reply:
This PR does not remove any keywords.

MySQL/MariaDB reserved keywords are newly hold in "reserved" private property and asserted with a test (in the `TokenizerTest`, we newly maintain a list for each DB vendor, so now and in the future we can easily and reliably maintain the platform-independent lists in the Tokenizer).

All other keywords, ie. not MySQL/MariaDB "reserved" keywords, are hold in "nonReserved" private property. For highlighting, the "reserved" and "nonReserved" keywords are used the same way, thus there is no functional change (between reserved/non-reserved keywords) on formatting nor highlighting.

With this said, this PR "effectively only adds missing keywords" and "improve tests". This is also observed in the updated highlighting functional tests.

Reply:
Are there keywords that could be reserved according to MySQL/MariaDB, but not according to other vendors?

Reply:
Currently highly possibly but let's merge this as a bugfix fixing the most missing keywords and later we can and should add more vendors into the `TokenizerTest` and assert such words are specified in `Tokenizer`.

Reply:
Why did you make me ask 3 times?

Reply:
I gave you my answer - Currently highly possibly.

If you want more in depth answer please extend your question or tell me your concerns.

Reply:
> I gave you my answer - Currently highly possibly.

Yes, you did give me that answer, but only after I asked 3 times. So, I will ask again:

Why did you make me ask 3 times?

Is that also a question you are going to make me ask 3 times?

Reply:
Please tell me what do to.

Reply:
When I ask a question, answer that question. Don't answer another question, or if you don't know, just say so. So for the third time, "Why did you make me ask 3 times?" I want a honest and plausible answer, is that too much to ask?

Reply:
I tried to answer your question in my 1st answer with some explanation...

So are we on the same page now? Is here anything else actionable I should amend in this PR?

Reply:
> I tried to answer your question in my 1st answer with some explanation...

Really? Let's see…

The question was:

> Are there keywords that could be reserved according to MySQL/MariaDB, but not according to other vendors?

You answered:

> The reasons are historical, jdorn library was based on MySQL keywords solely.

This feels more like an answer to the second question, which was:

>  I don't get why MySQL/MariaDB is privileged.


Then you went on with:

> I already added full SQLite reserved keywords list to the tests so all of them are newly contained in the "non-reserved" list.

Still aimed at the second question

> This PR aims to add MySQL/MariaDB/SQLite keywords. Other vendors can be added later.

Still aimed at the second question as well

> For highlighting, "reserved"/"non-reserved" lists are used the same way. For https://github.com/doctrine/sql-formatter/issues/32, we need "reserved" list to be minimal subset (intersect) of all vendors.

Interesting, but does not have to do with any of the 2 questions.

> So are we on the same page now?

Well… no! You said you tried to answer your question in your first answer, but I think I just demonstrated that you completely ignored it. It feels like we live in separate realities.

> Is here anything else actionable I should amend in this PR?

One thing at a time please. Before I invest more time in this PR, I'd like to understand if you are deliberately wasting my time or not.

Reply:
Your initial question is:

> When reading that, one might think they shouldn't add keywords for any other DB… is that the case?

I understand your question as concern about "beiing MySQL specific only". Such question is of course sensual as this lib aims to be DB vendor independent.

For a moment, let's dive into my motivation. In order to implement #32 correctly, we need to be sure what keywords are "reserved" and what are "non-reserved".

This is impossible to do trully for all DB vendors in the world, as anyone can come up with DB vendor with smaller subset.

So I created exact "reserved" keyword lists for 3 DB vendors (MySQL, MariaDB, SQLite) for start. This step is unambiguous.

Now here is the "unambiguous step". How to define the tokenizing lists of DB vendor independent `Tokenizer`\*?

-> This is my answer to your initial question - a question.

Now let me answer my PoV on my approach to that question.

For now, we use "reserved" and "non-reserved" the same way. So for now, we do not have to proove "all reserved" keywords are "reserved on all DB vendors".

With this important fact, I have taken the exact MySQL/MariaDB/SQLite lists and added all missing reserved keywords to the "reserved" keywords and everything else to the "non-reserved" keywords.

-> This is my approach.

Now let dicuss how to minimize the ambiguity. There are two solutions. One is to have a separate Tokenizer for each DB vendor. Another is to define limited supported DB vendors. This is actually not that easy, as each DB vendor is versioned and the grammar and keywords are not guaranteed to be not different across versions.

So, in my opinion, as long we do not want to maintain Tokenizer for each DB vendor and version, we need to assert these two things:
- allow syntax fomatting/highlighting to be "mostly best looking over always logical" - this have no impact on DB results
- minimize SQL changes that have impact on DB results - as changing case of "keyword token" has no impact only when such keyword is "reserved", we must define the "reserved" keyword list to be a subset/intersect of "reserved" keywords for all supported DB vendors.

-> This is how the lists should be maintained and to what definition they should converge in the future.

So yes, no the list in Tokenizer should be made "for any specific DB vendor" even if this is actually how the lib is coded now.

I propose to rename the property desctiption to not mention anything about MySQL (or any other DB vendor).

Then I need to know exactly what exactly is fine/problematic on this PR so I can put this PR into that state and possibly solve the rest in another PR. Maybe keep focus in this PR to add the DB vendor specific lists in the `TokenizerTest` plus assert all such keywords are present as "reserved" in `Tokenizer` and discuss the distinguish between "reserved" vs "non-reserved" compilation in another PR?

This is actionable and unambiguous. Ok?

\* we cannot detect used DB vendor reliably, as `SELECT 1 BLOB` can be invalid SQL on one DB vendor and valid on another, as `BLOB` may or may not be "reserved" keyword.

Reply:
@greg0ire Thank you your feedback helped me to realize few important facts and I have reworked this PR.

I have removed the ambiguous "reserved" vs "non-reserved" categorization and added exact "reserved" keyword list for each major DB vendor.

Let me please know what you think and if I should possibly extract some commit into separate PR.